src/foam/dao/MDAO.java

```diff
@@ -80,11 +80,15 @@ synchronized void setState(Object state) {
   }
 
   public FObject objIn(FObject obj) {
-    return obj.fclone().freeze();
+    FObject o = obj.fclone();
```

Comment:
Because the javaGetter wants to use getX(), we can't lose the context on object in and out of the MDAO. This feels bad but I'm not sure what other path to take.